### PR TITLE
Replace mnt-rs with libmount

### PR DESCRIPTION
**Pull request by tasleson**
The library `mnt-rs` utilzes `/proc/mounts` which contains inaccurate information and caused issue https://github.com/stratis-storage/stratisd/issues/695.  After doing some research, it appears that `/proc/self/mountinfo` is the correct place to look for mount information, ref. https://bugzilla.redhat.com/show_bug.cgi?id=491924 .  @agrover did a PR to utilize `/proc/self/mountinfo` to `mnt-rs` ref. https://github.com/stemjail/mnt-rs/pull/4 , but that PR has stalled.  Thus the decision was made to utilize `libmount` instead as it does utilize `/proc/self/mountinfo`.

### CODE REVIEW

**Comment by tasleson** [comment]
Rebased

File: src/engine/strat_engine/thinpool/filesystem.rs [hunk 1]
```diff
@@ -157,27 +159,18 @@ impl StratFilesystem {
...
+        let parser = libmount::mountinfo::Parser::new(mount_data.as_bytes());
 
         let mut last_error: Option<String> = None;
-        for mp in m_iter {
+        for mp in parser.into_iter() {
             match mp {
                 Ok(mount) => {
-                    if mount.contains(&MountParam::Spec(search)) {
-                        return Ok(Some(mount.file));
+                    if mount.mount_source == *search {
```
Comment (agrover) [inline src/engine/strat_engine/thinpool/filesystem.rs]: Yeah, I think we just want to use maj/min for comparison here, is all. Something like:
```
let major = self.thin_dev.device().major as u64;
let minor = self.thin_dev.device().minor as u64;
if mount.major == major && mount.minor == minor {
```

**Comment by agrover** [comment]
Why 060326b? Aren't mount.major and .minor already u64?

**Comment by tasleson** [comment]
@agrover 
> Why 060326b? Aren't mount.major and .minor already u64?

```rust
pub struct Device {
    /// Device major number
    pub major: u32,
    /// Device minor number
    pub minor: u32,
}
```
Apparently `devicemapper-0.21.0/src/device.rs` has them as `u32`

**Comment by tasleson** [comment]
Fixes https://github.com/stratis-storage/stratisd/issues/820

**Comment by agrover** [comment]
ok thanks. One of THOSE situations.

:+1:  once Clippy is made happy.

**Comment by agrover** [comment]
fwiw @tasleson you can find/copy the original problem description in #695.

**Comment by tasleson** [comment]
@mulkieran @agrover  Please see updated PR message.  Hopefully this clears things up.

**Comment by mulkieran** [comment]
A 21 comment bz from 2009 in which the last entry is a promise to fix this once and for all in util-linux couldn't really be said to clear anything up :( But I guess it's a starting point for further research if necessary.

**Comment by mulkieran** [comment]
I expect there will be some more minor clippy errors, but once those are fixed, I'm fine w/ it.

**Comment by mulkieran** [comment]
Except that you might as well close this and start a new PR that squashes the commits that should be squashed, like the clippy fixes.

**Comment by tasleson** [comment]
Passing all checks, I squashed, should be able to merge when CI is done

**Pull request merged**